File: app/src/main/java/com/ihadzhi/eatlimination/data/DietDao.java
package com.ihadzhi.eatlimination.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface DietDao {

    @Query("SELECT * FROM Diet WHERE active > 0 ORDER BY createdAt DESC")
    LiveData<Diet> fetchActiveDiet();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(Diet diet);

    @Delete
    void delete(Diet diet);

}

File: app/src/main/java/com/ihadzhi/eatlimination/data/DietDao.kt
package com.ihadzhi.eatlimination.data

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface DietDao {
    @Query("SELECT * FROM Diet WHERE isActive > 0 ORDER BY createdAt DESC")
    fun fetchActiveDiet(): LiveData<Diet>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(diet: Diet?): Long

    @Delete
    fun delete(diet: Diet?)
}